File: src/main/java/net/adventuregame/game/GameLoop.java
// GameLoop.java
package net.adventuregame.game;

import com.chidozie.core.renderEngine.WindowManager;
import net.adventuregame.story.StoryHandler;

public class GameLoop {
    private final GameState state;
    private final WindowManager window;

    public GameLoop(GameState state) {
        this.state = state;
        this.window = state.window;
    }

    public void run() {
        while (!window.windowShouldClose()) {
            state.update();
            state.render();
            window.update();
        }
        state.cleanup();
    }
}

File: src/main/kotlin/net/adventuregame/game/GameLoop.kt
// GameLoop.java
package net.adventuregame.game

import com.adv.core.renderEngine.WindowManager
import net.adventuregame.processLuaMainThreadQueue

class GameLoop(state: GameState) {
    private val state: GameState?
    private val window: WindowManager

    init {
        this.state = state
        this.window = state.window
    }

    fun run() {
        while (!window.windowShouldClose()) {
            state!!.update()
            processLuaMainThreadQueue()
            state.render()
            window.update()
        }
        state!!.cleanup()
    }
}
